rpc/query_server.go

```diff
@@ -1073,6 +1082,79 @@ func (s *QueryServer) EthAccounts() ([]eth.Data, error) {
 	return []eth.Data{}, nil
 }
 
+func (s *QueryServer) GetValidators() (*blockatlas.JsonGetValidators, error) {
+	snapshot := s.StateProvider.ReadOnlyState()
+	defer snapshot.Release()
+
+	ctx, err := s.createDPOS3Ctx(snapshot)
+	if err != nil {
+		return nil, err
+	}
+
+	validators, err := dposv3.ValidatorList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	candidates, err := dposv3.LoadCandidateList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	getValidatorResponse := make([]blockatlas.Validator, 0)
+	chainID := ctx.Block().ChainID
+
+	for _, validator := range validators {
```

Comment:
This loop has time complexity equal to n^2. We can do it better by using hash map.

First we loop to save validator address in hash map like
```
validatorList := map[string]loom.Address
for _, validator := range validators {
  validatorList[string(validator.Pubkey)] = loom.Address{ChainID: chainID, Local: loom.LocalAddressFromPublicKey(validator.PubKey)}
}
```
and then we can just iterate candidates that are in the validators list